### DISCUSSION TITLE
Invoice PDF Header Margin configuration

### DISCUSSION THREAD

**Discussion by lamp4youde** [kimai/kimai#2279] [Questions & Answers]
Hi,

as far as I found out, PDF invoice header margin is hard coded in 

src/Invoice/Renderer/PdfRenderer.php

        $content = $this->converter->convertToPdf($content, [
            'setAutoTopMargin' => 'pad',
            'setAutoBottomMargin' => 'pad',
            'margin_top' => 12,
            'margin_bottom' => 8,
    ]);

But I haven't found out how I configure this in my twig template.

For now I changed the PdfRenderer.php - but I think the next update will overwrite my changes.

Hope someone can help...

**Comment by kevinpapst** [comment]
Try to use the [CSS Paged Media via @page](https://mpdf.github.io/paging/using-page.html):
```
    @page {
        size: auto;
        margin-top: 35mm;
        margin-right: 15mm;
        margin-bottom: 35mm;
        margin-left: 15mm;
    }
```
The default values in `PdfRenderer.php` don't belong there, but now its too late to change it as templates out there need them...
